src/RecursionChecker.php

```diff
@@ -96,14 +96,16 @@ private function checkClassMetadata(ClassMetadata $classMetadata, RecursionConte
                     }
                 }
 
-                /*
-                 * Future feature idea: Instead of 2, we could use a configurable maximum recursion depth.
-                 * One could then allow to unroll a recursion up to a certain number and stop it after that.
-                 */
-                if ($propertyContext->countClassNames($propertyClassMetadata->getClassName()) > 2) {
+                $maxDepth = $property->getMaxDepth();
+                $stackCount = $context->countClassNames($classMetadata->getClassName());
```

Comment:
you switch from propertyContext to context, meaning the current property is not yet pushed. unless i am confused, this would make the $stackCount one less than before, or not?